### PR TITLE
14days

### PR DESCRIPTION
**Pull request by at-huyennguyen**
paginate

### CODE REVIEW

File: CRUDDemo/app/controllers/paginate_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+class PaginateController < ApplicationController
+ @@per = 50
+  def index
+    @select =[10,20,50]
+    @examples = ExampleModel.all
+    @@per = params[:limit_selection] ||= @select[0]
+    @page_number = @examples.length/@@per.to_i
```
Comment (at-longhoang) [inline CRUDDemo/app/controllers/paginate_controller.rb]: move to new method

File: CRUDDemo/app/controllers/paginate_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+class PaginateController < ApplicationController
+ @@per = 50
```
Comment (at-longhoang) [inline CRUDDemo/app/controllers/paginate_controller.rb]: use instance var

File: CRUDDemo/app/controllers/paginate_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  def index
+    @select =[10,20,50]
+    @examples = ExampleModel.all
+    @@per = params[:limit_selection] ||= @select[0]
+    @page_number = @examples.length/@@per.to_i
+    @current_page =params[:current_page]||= 1
+    offset_value = (@current_page.to_i == 1) ? 0 : ((@current_page.to_i-1)*@@per.to_i)
```
Comment (at-longhoang) [inline CRUDDemo/app/controllers/paginate_controller.rb]: split into new method

File: CRUDDemo/app/helpers/paginate_helper.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module PaginateHelper
+  
+  # def test
```
Comment (at-longhoang) [inline CRUDDemo/app/helpers/paginate_helper.rb]: remove if you dont use

File: CRUDDemo/app/views/paginate/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+<table>
+	<thead>
+		<tr>
+		  <td>No</td>
+		  <td>Name</td>
+		  <td>Phone</td>
+		  <td>Address</td>
+		</tr>
+	</thead>
+	<tbody>
```
Comment (at-longhoang) [inline CRUDDemo/app/views/paginate/index.html.erb]: Fix indent

File: test.xt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    @select =[10,20,50]
+    @examples = ExampleModel.all
+   
+    $per = params[:limit_tag] ||= @select[0]
+
+    @page_number = @examples.length/$per.to_i
+    @page =params[:current_page] ||= params[:page] ||= 1
+    offset_value = (@page.to_i == 1) ? 0 : ((@page.to_i-1)*$per)
+    @recent_views = @examples.per_page(offset_value)
+    binding.pry
```
Comment (at-longhoang) [inline test.xt]: remove binding.pry

Comment (at-huyennguyen) [inline test.xt]: just a test file, I'll remove it :)

File: CRUDDemo/app/controllers/paginate_controller.rb (hunk 2 above)
Comment (at-huyennguyen) [inline CRUDDemo/app/controllers/paginate_controller.rb]: fixed

File: CRUDDemo/app/controllers/paginate_controller.rb (hunk 1 above)
Comment (at-huyennguyen) [inline CRUDDemo/app/controllers/paginate_controller.rb]: fixed

File: CRUDDemo/app/controllers/paginate_controller.rb (hunk 3 above)
Comment (at-huyennguyen) [inline CRUDDemo/app/controllers/paginate_controller.rb]: fixed

**Pull request merged**